data_structures/src/serializers.rs

```diff
@@ -0,0 +1,572 @@
+                    builder,
+                    &AddressArgs {
+                        ip_type: FlatBuffersIpAddress::Ipv4,
+                        ip: Some(ip_command.as_union_value()),
+                        port: peer.port,
+                    },
+                )
+            }
+            WitnetIpv6 { ip0, ip1, ip2, ip3 } => {
+                let ip_command =
+                    FlatBuffersIpv6::create(builder, &FlatBuffersIpv6Args { ip0, ip1, ip2, ip3 });
+                FlatBufferAddress::create(
+                    builder,
+                    &AddressArgs {
+                        ip_type: FlatBuffersIpAddress::Ipv6,
+                        ip: Some(ip_command.as_union_value()),
+                        port: peer.port,
+                    },
+                )
+            }
+        })
+        .collect();
+
+    let addresses = Some(builder.create_vector(&addresses_command));
+    let peers_command = FlatBuffersPeers::create(builder, &PeersArgs { peers: addresses });
+
+    let message = FlatBuffersMessage::create(
+        builder,
+        &MessageArgs {
+            magic: peers_args.magic,
+            command_type: FlatBuffersCommand::Peers,
+            command: Some(peers_command.as_union_value()),
+        },
+    );
+    build_flatbuffer(builder, message)
+}
+
+// Create a witnet's peers message to decode a flatbuffers' peers message
+fn create_peers_message(peers_args: PeersWitnetArgs) -> Option<Message> {
+    peers_args.peers.peers().map(|ftb_addresses| {
+        // TODO: Refactor as declarative code [24-10-2018]
+        let len = ftb_addresses.len();
+        let mut counter = 0;
+        let mut ftb_address: Option<Address>;
+        let mut peer;
+        let mut vec_addresses = Vec::new();
+        while counter < len {
+            peer = ftb_addresses.get(counter);
+            ftb_address = create_address(peer);
+            if ftb_address.is_some() {
+                vec_addresses.push(ftb_address.unwrap());
+            }
+            counter += 1;
+        }
+        Message {
+            kind: Command::Peers {
+                peers: vec_addresses,
+            },
+            magic: peers_args.magic,
+        }
+    })
+}
+
+// Create a ping flatbuffer to encode a witnet's ping message
+fn create_ping_flatbuffer(
+    builder: &mut FlatBufferBuilder,
+    ping_args: PingFlatbufferArgs,
+) -> Vec<u8> {
+    let ping_command = FlatBuffersPing::create(
+        builder,
+        &PingArgs {
+            nonce: ping_args.nonce.to_owned(),
+        },
+    );
+    let message = FlatBuffersMessage::create(
+        builder,
+        &MessageArgs {
+            magic: ping_args.magic,
+            command_type: FlatBuffersCommand::Ping,
+            command: Some(ping_command.as_union_value()),
+        },
+    );
+
+    build_flatbuffer(builder, message)
+}
+
+// Create a witnet's ping message to decode a flatbuffers' ping message
+fn create_ping_message(ping_args: &PingWitnetArgs) -> Message {
```

Comment:
I would make this function take the ownership of its parameter. Similar functions like `create_get_peers_message` and `create_verack_message` already do it like that. To me, it makes more sense as this function transforms its input parameters into another different thing and returns it.

```suggestion
fn create_ping_message(ping_args: PingWitnetArgs) -> Message {
```

Reply:
But if you want to borrow the parameter it's ok by me, as long all similar functions are consistent in their taking the ownership/borrowing policy